BrainPortal/app/models/s3_flat_data_provider.rb

```diff
@@ -0,0 +1,293 @@
+
+#
+# CBRAIN Project
+#
+# Copyright (C) 2008-2012
+# The Royal Institution for the Advancement of Learning
+# McGill University
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+# Implements a DataProvider that stores its files
+# in a Amazon S3 bucket. Single files are saved
+# as-is, FileCollections are tar'ed and untar'ed
+# as needed.
+class S3FlatDataProvider < DataProvider
+
+  Revision_info=CbrainFileRevision[__FILE__] #:nodoc:
+
+  validates_presence_of :cloud_storage_client_identifier, :cloud_storage_client_token,
+                        :cloud_storage_client_path_start, :cloud_storage_client_bucket_name
+
+  attr_accessor :s3_connection
```

Comment:
You define an accessor method for the variable, yet everywhere you use the instance variable directly, EXCEPT at lines 97 and 101. Chose one or the other, but not a mix... (I suggest removing the accessor completely, unless you code external to this class will need to access it).